src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1692,9 +1692,11 @@ public static void Sort(Array keys, Array? items, int index, int length, ICompar
                             GenericSort<double>(keys, items, adjustedIndex, length);
                             return;
                         case CorElementType.ELEMENT_TYPE_I:
+                            GenericSort<IntPtr>(keys, items, adjustedIndex, length);
+                            return;
                         case CorElementType.ELEMENT_TYPE_U:
```

Comment:
I think we can use the same trick as was done for `Reverse` here, that is treat `ELEMENT_TYPE_I` as `I4` on 32-bit and as `I8` on 64-bit, so we can avoid the additional generic instantiation.